### PR TITLE
Fix compatibility with Zeitwerk

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: lib/foreman_kubevirt/engine.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Engine < ::Rails::Engine
     config.autoload_paths += Dir["#{config.root}/app/models/concerns"]
     initializer "foreman_kubevirt.register_plugin", :before => :finisher_hook do |_app|
       Foreman::Plugin.register :foreman_kubevirt do
```
Comment (ofedoren) [inline lib/foreman_kubevirt/engine.rb:6]: Just to be aligned with other plugins, we could probably wrap this in `app.reloader.to_prepare` block.

**Review by ofedoren — COMMENTED** [review]
Thanks, @ekohl, I thought this plugin is safe, since https://github.com/theforeman/foreman_kubevirt/pull/158 didn't show any errors.

Now, just to double check, could you please add a commit to test with https://github.com/theforeman/foreman/pull/10299?

File: lib/foreman_kubevirt/engine.rb (hunk 1 above)
Comment (ekohl) [reply lib/foreman_kubevirt/engine.rb:6]: Perhaps that also solves the deprecation warning I still saw. Will try

Comment (ekohl) [reply lib/foreman_kubevirt/engine.rb:6]: Could you submit that as a change in https://github.com/theforeman/foreman_plugin_template/blob/master/lib/foreman_plugin_template/engine.rb as well?

Comment (ekohl) [reply lib/foreman_kubevirt/engine.rb]: > Perhaps that also solves the deprecation warning I still saw. Will try

That it does.

**Comment by ekohl** [comment]
It's green both with and without:
image

**Review by ofedoren — APPROVED** [review]
Still weird it didn't fail before, but now it at least seems safer. Thanks, @ekohl ! Don't forget to drop the last commit before merge :)

**Review by evgeni — APPROVED** [review]

File: lib/foreman_kubevirt/engine.rb (hunk 1 above)
Comment (ofedoren) [reply lib/foreman_kubevirt/engine.rb]: > Could you submit that as a change in https://github.com/theforeman/foreman_plugin_template/blob/master/lib/foreman_plugin_template/engine.rb as well?

https://github.com/theforeman/foreman_plugin_template/pull/67

**Comment by ekohl** [comment]
Waiting for https://github.com/theforeman/foreman_kubevirt/pull/165, marking as draft to be sure.

**Comment by ekohl** [comment]
@ofedoren does it look good to you?

**Review by ofedoren — APPROVED** [review]
Green and beautiful :)

**Review by adamruzicka — APPROVED** [review]

**Pull request merged**